pkg/crypto/certloader/reloader.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package certloader
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+// Reloader is a set of TLS configuration files including custom CAs, and a
+// certificate along with its private key (keypair) that can be reloaded
+// dynamically via the Reload* functions.
+type Reloader interface {
```

Comment:
nit: This interface seems unused. I'd prefer to remove it until we actually need it.